contracts/Uni.sol

```diff
@@ -110,6 +110,23 @@ contract Uni {
         _moveDelegates(address(0), delegates[dst], amount);
     }
 
+    /**
+     * @notice Burns tokens
+     * @param rawAmount The number of tokens to be minted
+     */
+    function burn(uint rawAmount) external {
+        uint96 amount = safe96(rawAmount, "Vexchange::approve: amount exceeds 96 bits");
+        require(amount <= balances[msg.sender], "Vexchange::burn token balance is less than the burn amount");
+
+        balances[msg.sender] = sub96(balances[msg.sender], amount, "Vexchange::burn new balance underflows");
+        totalSupply = safe96(SafeMath.sub(totalSupply, amount), "Vexchange::burn new supply exceeds 96 bits");
```

Comment:
Interestingly, we have two actions being taken here:
 - result = SafeMath.sub(totalSupply, amount)
 - safe96(result)

safe96 should never fail but comes with an error message, SafeMath.sub should also never fail but doesn't come with an error message. Unsure if any actions needs to be taken here or not.

Reply:
Same thing here, I was aware and just wanted to be extra safe. We can do away with the safe96 call and the SafeMath.sub operation alone should suffice